pyproject.toml

```diff
@@ -110,7 +110,7 @@ full = [
 
 # Dependencies for running the test infrastructure
 test = [
-    "pytest!=8.0.0rc1,!=8.0.0rc2",
+    "pytest>=8.0.0rc2",
```

Comment:
@drammock we have to do this if we want to do:
```
with pytest.warns(...), pytest.warns(...):
```
because on `<8` the inner context manager won't reemit the warnings but on `>=8` it will.

Alternatively I could change all instances to `with _record_warnings(), pytest.warns(...)` (i.e., a single `pytest.warns` being caught) but then we lose the option to have extra specificity.

I don't love either option really... a third option would be to have a `with also_warns(...), pytest.warns(...)` where `also_warns == pytest.warns` for 8+ and `also_warns == _record_warnings` (i.e., noop) when on `<8`. But this is custom code again, at least for a while until we *do* want to pin to 8+ then we can substitute `pytest.warns` for all `second_warns`. Maybe that's the safest option here?